spec/macros/duke/urls_spec.rb

```diff
@@ -52,6 +59,67 @@
       end
     end
 
+    context 'DUKE-AK-445-use-941-fields' do
+      context 'Record with one(1) journal portfolio' do
+        it "sets a 'url' with an href that includes 'soa_url'" do
+          parsed_url = JSON.parse(iee_journal_rec['url'][0])
+          expect(parsed_url['href']).to include(soa_url)
+        end
+
+        it "expects 'url' to include the record's ALMA number" do
+          alma_number = JSON.parse(iee_journal_rec['alma_number']).to_s
+          parsed_url = JSON.parse(iee_journal_rec['url'][0])
+          expect(parsed_url['href']).to include(alma_number)
+        end
+      end
+
+      context 'Record with only multiple journal portfolios' do
+        it 'sets a `url` with an href that includes `soa_url`' do
+          parsed_url = JSON.parse(iee_multi_journals_rec['url'][0])
+          expect(parsed_url['href']).to include(soa_url)
+        end
+
+        it "expects `url` to include the record's ALMA number" do
+          alma_number = JSON.parse(iee_multi_journals_rec['alma_number']).to_s
+          parsed_url = JSON.parse(iee_multi_journals_rec['url'][0])
+          expect(parsed_url['href']).to include(alma_number)
+        end
+      end
+
+      context 'Record with one(1) non-journal portfolio' do
+        it 'sets a `url` href that uses value from 943d field' do
+          parsed_url = JSON.parse(iee_nonjournal_rec['url'][0])
+          expect(parsed_url['href']).not_to include(soa_url)
```

Comment:
Can we add a more specific expectation here that more clearly shows that the `href` matches the `943d`? As written, the test would still pass even if the final extracted href were an empty string. E.g.
```
expect(parsed_url['href']).to eq('https://na05.alma.exlibrisgroup.com/view/uresolver/01DUKE_INST/openurl?u.ignore_date_coverage=true&amp;portfolio_pid=53891453500008501&amp;Force_direct=true')
```